src/proc/formats-converter.cpp

```diff
@@ -35,26 +67,32 @@ stream_profiles formats_converter::get_all_possible_profiles( const stream_profi
 
     for( auto & raw_profile : raw_profiles )
     {
-        for( auto && pbf : _pb_factories )
+        for( auto & pbf : _pb_factories )
         {
-            const auto && sources = pbf->get_source_info();
-            for( auto && source : sources )
+            const auto & sources = pbf->get_source_info();
```

Comment:
If you want a reference, I think we should actually change `processing_block_factory::get_source_info()` and `::get_target_info()` to return references.

Reply:
Updated `processing_block_factory::get_source/target_info()` to return a reference. The factory and formats_converter are the only users.